nilearn/mass_univariate/permuted_least_squares.py

```diff
@@ -470,43 +715,155 @@ def permuted_ols(tested_vars, target_vars, confounding_vars=None,
-                      'be running. You may want to perform more permutations '
-                      'in order to take the most of the available computing '
-                      'ressources.' % (n_perm, n_jobs, n_perm))
+        warnings.warn(
+            f'The specified number of permutations is {n_perm} and the number '
+            f'of jobs to be performed in parallel has set to {n_jobs}. '
+            f'This is incompatible so only {n_perm} jobs will be running. '
+            'You may want to perform more permutations in order to take the '
+            'most of the available computing ressources.'
+        )
         n_perm_chunks = np.ones(n_perm, dtype=int)
+
     else:  # 0 or negative number of permutations => original data scores only
         if two_sided_test:
             scores_original_data = (scores_original_data
                                     * sign_scores_original_data)
+
         return np.asarray([]), scores_original_data.T, np.asarray([])
 
     # actual permutations, seeded from a random integer between 0 and maximum
     # value represented by np.int32 (to have a large entropy).
     ret = joblib.Parallel(n_jobs=n_jobs, verbose=verbose)(
         joblib.delayed(_permuted_ols_on_chunk)(
-            scores_original_data, testedvars_resid_covars,
-            targetvars_resid_covars.T, thread_id + 1, covars_orthonormalized,
-            n_perm=n_perm, n_perm_chunk=n_perm_chunk,
-            intercept_test=intercept_test, two_sided_test=two_sided_test,
+            scores_original_data,
+            testedvars_resid_covars,
+            targetvars_resid_covars.T,
+            thread_id + 1,
+            covars_orthonormalized,
+            masker=masker,
+            cdt=cdt,
+            n_perm=n_perm,
+            n_perm_chunk=n_perm_chunk,
+            intercept_test=intercept_test,
+            two_sided_test=two_sided_test,
             random_state=rng.randint(1, np.iinfo(np.int32).max - 1),
-            verbose=verbose)
+            verbose=verbose,
+        )
         for thread_id, n_perm_chunk in enumerate(n_perm_chunks))
+
     # reduce results
-    scores_as_ranks_parts, h0_fmax_parts = zip(*ret)
-    h0_fmax = np.hstack((h0_fmax_parts))
-    scores_as_ranks = np.zeros((n_regressors, n_descriptors))
-    for scores_as_ranks_part in scores_as_ranks_parts:
-        scores_as_ranks += scores_as_ranks_part
-    # convert ranks into p-values
-    pvals = (n_perm + 1 - scores_as_ranks) / float(1 + n_perm)
+    (
+        vfwe_scores_as_ranks_parts,
+        vfwe_h0_parts,
+        csfwe_h0_parts,
+        cmfwe_h0_parts,
+    ) = zip(*ret)
+
+    # Voxel-level FWE
+    vfwe_h0 = np.hstack((vfwe_h0_parts))
+    vfwe_scores_as_ranks = np.zeros((n_regressors, n_descriptors))
+    for vfwe_scores_as_ranks_part in vfwe_scores_as_ranks_parts:
+        vfwe_scores_as_ranks += vfwe_scores_as_ranks_part
+
+    vfwe_pvals = (n_perm + 1 - vfwe_scores_as_ranks) / float(1 + n_perm)
```

Comment:
The generic "pvals" from before are now specifically referred to as "vfwe_pvals" to distinguish them from the other corrected p-value outputs.